Gems/Atom/RPI/Code/Source/RPI.Reflect/Model/ModelAssetHelpers.cpp

```diff
@@ -0,0 +1,206 @@
+ */
+
+#include <Atom/RPI.Reflect/Buffer/BufferAssetCreator.h>
+#include <Atom/RPI.Reflect/Model/ModelAssetHelpers.h>
+#include <Atom/RPI.Reflect/Model/ModelLodAssetCreator.h>
+#include <Atom/RPI.Reflect/ResourcePoolAssetCreator.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        // Build a buffer asset that contains the given data.
+        AZ::Data::Asset<AZ::RPI::BufferAsset> ModelAssetHelpers::CreateBufferAsset(
+            const void* data, const uint32_t elementCount, const uint32_t elementSize)
+        {
+            // Create a buffer pool asset for use with the buffer asset
+            AZ::Data::Asset<AZ::RPI::ResourcePoolAsset> bufferPoolAsset;
+            {
+                AZ::Data::AssetId bufferPoolId = AZ::Uuid::CreateRandom();
+                bufferPoolAsset = AZ::Data::AssetManager::Instance().CreateAsset(
+                    bufferPoolId, azrtti_typeid<AZ::RPI::ResourcePoolAsset>(), Data::AssetLoadBehavior::PreLoad);
+
+                auto bufferPoolDesc = AZStd::make_unique<AZ::RHI::BufferPoolDescriptor>();
+                bufferPoolDesc->m_bindFlags = AZ::RHI::BufferBindFlags::InputAssembly;
+                bufferPoolDesc->m_heapMemoryLevel = AZ::RHI::HeapMemoryLevel::Host;
+
+                AZ::RPI::ResourcePoolAssetCreator creator;
+                creator.Begin(bufferPoolId);
+                creator.SetPoolDescriptor(AZStd::move(bufferPoolDesc));
+                creator.SetPoolName("ModelAssetHelperBufferPool");
+                creator.End(bufferPoolAsset);
+            }
+
+            // Create a buffer asset that contains a copy of the input data.
+            AZ::Data::Asset<AZ::RPI::BufferAsset> asset;
+            {
+                AZ::Data::AssetId bufferId = AZ::Uuid::CreateRandom();
+                asset = AZ::Data::AssetManager::Instance().CreateAsset(
+                    bufferId, azrtti_typeid<AZ::RPI::BufferAsset>(), Data::AssetLoadBehavior::PreLoad);
+
+                AZ::RHI::BufferDescriptor bufferDescriptor;
+                bufferDescriptor.m_bindFlags = AZ::RHI::BufferBindFlags::InputAssembly;
+                bufferDescriptor.m_byteCount = elementCount * elementSize;
+
+                AZ::RPI::BufferAssetCreator creator;
+                creator.Begin(bufferId);
+                creator.SetPoolAsset(bufferPoolAsset);
+                creator.SetBuffer(data, bufferDescriptor.m_byteCount, bufferDescriptor);
+                creator.SetBufferViewDescriptor(AZ::RHI::BufferViewDescriptor::CreateStructured(0, elementCount, elementSize));
+                creator.End(asset);
+            }
+
+            return asset;
+        }
+
+        void ModelAssetHelpers::CreateUnitCube(ModelAsset* modelAsset)
+        {
+            // Build a mesh containing a unit cube.
+            // The vertices are duplicated for each face so that we can have correct per-face normals and UVs.
+
+            // 6 vertices per face, 6 faces.
+            constexpr AZStd::array<uint32_t, 6 * 6> indices = {
```

Comment:
I'm wondering if there might be a simpler to create a single face struct and transform it 6 times to generate the values for all the buffers.  Kinda like the icosphere version https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Geometry3DUtils.cpp#L51 e.g.
```c++
// a single forward facing quad
struct quad {
    float positions[6] = {};
    float normal[3] = {0,1,0};
    uint32_t indicies[6] = {0,1,2,0,2,3};
    float uvs[2 * 4] = {0.0f, 0.0f, 1.0f, 0.0f, 1.0f, 1.0f, 0.0f, 1.0f};
    float tangent[4] = {0.0f, 0.0f, 1.0f, 1.0f};
    float bitangent[4] = {1.0f, +0.0f, +0.0f, +1.0f}
};

// reserve 
indices.reserve(blah);
positions.reserve(blah);

for(faceRotation : faceRotations)
{
     // rotate face and fill in buffers
}
```

Reply:
Probably. I was thinking about that too, especially now that I've added a second version that generates an X instead of a cube based on UX feedback (they were concerned the cube would look too correct if viewed in-scene with a material on it). 

On the other hand, the version I have is still pretty straightforward, I'm not fully convinced a data generation version of the function will actually be smaller or simpler. If you want to refactor it in a follow-up PR I'd be fully supportive of that though. ;)